docs/terminals.md

```diff
@@ -0,0 +1,141 @@
+# Terminals and Standard IO
+All processes on Unix (and Unix-like) operating systems have 3 standard file descriptors (fd) passed to them at start, collectively standard-IO (`stdio`):
+
+* `0`: standard-in (`stdin`), the input stream into the process
+* `1`: standard-out (`stdout`), the output stream from the process
+* `2`: standard-error (`stderr`), the error stream from the process
```

Comment:
I would recommend at least mentioning `--preserve-fds` here, with the note that it is **separate** to all of this handling for `stdio` and that those file descriptors are passed verbatim in all cases.

Reply:
OK. I just added something at the end.